contrib/CONTRIBUTING.md

```diff
@@ -0,0 +1,232 @@
+1. [How Can I Contribute?](#how-can-i-contribute)
+   1. [Code Contribution General Guideline](#code-contribution-general-guidelines)
+   1. [Pull Request Philosophy](#pull-request-philosophy)
+   1. [Pull Request Process](#pull-request-process)
+   1. [Testing](#testing)
+   1. [Addressing Feedback](#addressing-feedback)
+   1. [Squashing Commits](#squashing-commits)
+   1. [Refactoring](#refactoring)
+ 1. [Reporting Bugs](#reporting-bugs)
+ 1. [Suggesting Features](#suggesting-enhancements)
+
+
+## I don't want to read this whole thing I just have a question!
+
+> **Note:** Please don't file an issue to ask a question. You'll get faster results by using the resources below.
+
+We have an official Discord server where the community chimes in with helpful advice if you have questions. 
+This is the fastest way to get an answer and the core development team is active on Discord.
+
+* [Official Bittensor Discord](https://discord.gg/7wvFuPJZgq)
+
+## What should I know before I get started?
+Bittensor is still in the Alpha stages, and as such you will likely run into some problems in deploying your model or installing Bittensor itself. If you run into an issue or end up resolving an issue yourself, feel free to create a pull request with a fix or with a fix to the documentation. The documentation repository can be found [here](https://github.com/opentensor/docs). 
+
+Additionally, note that the core implementation of Bittensor consists of two separate repositories: [The core Bittensor code](https://github.com/opentensor/bittensor) and the Bittensor Blockchain [subtensor](https://github.com/opentensor/subtensor).
+
+Supplemental, yet necessary repositories are [openvalidators](https://github.com/opentensor/validators) and [openminers](https://github.com/opentensor/miners).
+
+## How Can I Contribute?
+
+You can contribute to Bittensor in one of three main ways (as well as many others):
+1. [Bug](#reporting-bugs) reporting and fixes
+2. New features and Bittensor [enhancements](#suggesting-enhancements)
+3. [Documentation](https://github.com/opentensor/docs) fixes and updates
+
+> Please follow the Bittensor [style guide](./STYLE.md) regardless of your contribution type. 
+
+Here is a high-level summary:
+- Code consistency is crucial; adhere to established programming language conventions.
+- Use `black` to format your Python code; it ensures readability and consistency.
+- Write concise Git commit messages; summarize changes in ~50 characters.
+- Follow these six commit rules:
+  - Atomic Commits: Focus on one task or fix per commit.
+  - Subject and Body Separation: Use a blank line to separate the subject from the body.
+  - Subject Line Length: Keep it under 50 characters for readability.
+  - Imperative Mood: Write subject line as if giving a command or instruction.
+  - Body Text Width: Wrap text manually at 72 characters.
+  - Body Content: Explain what changed and why, not how.
```

Comment:
I think some examples of commits that follow those rules and commits that don't would be very helpful for people who don't have much experience with git.

What do you think about adding some examples?

Reply:
Yes agreed.  They are in `STYLE.md`. I have linked for clarity rather than duplicate in two places.